src/ffi/dial_ffi.rs

```diff
@@ -205,10 +197,17 @@ pub unsafe extern "C" fn dial(
                     .connect()
                     .await?
             }
-            _ => {
+            (None, None) => {
                 let c = dial_without_cred(uri_str, allow_insec, disable_webrtc)?;
                 c.connect().await?
-                
+            }
+            (None, Some(_)) => {
+                // the return type here is finicky, so we are taking advantage of the ? operator to return an errenous value
```

Comment:
I know I suggested a comment like this, but on reflection I think it's a bit awkward and unclear. I think a better pattern would be to not use the `?` operator inside the `match` statement, but rather immediately outside of it.